File: LaserBotController/Point.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LaserBotController
{
	class Point
	{
		public double X { get; set; }
		public double Y { get; set; }

		public int X_i { get { return (int)Math.Round(X); } }
		public int Y_i { get { return (int)Math.Round(Y); } }

		public Point(double X, double Y)
		{
			this.X = X;
			this.Y = Y;
		}

		public void Change(double X, double Y)
		{
			this.X = X;
			this.Y = Y;
		}

		public void CheckBound(Point limitPoint)
		{
			if (X < Global.ZeroPoint.X || Y < Global.ZeroPoint.Y || X > limitPoint.X || Y > limitPoint.Y)
			{
				throw new Exception(String.Format("Point ({0}, {1}) exceed limit !", X, Y));
			}
		}

		public Point Copy()
		{
			return new Point(X, Y);
		}

		public bool Equals(Point p)
		{
			return p != null && X == p.X && Y == p.Y;
		}

		public override string ToString()
		{
			return "{" + X + ", " + Y + "}";
		}

		public double Distance(Point p)
		{
			return Math.Sqrt(Global.Pow2(X - p.X) + Global.Pow2(Y - p.Y));
		}
	}

}

File: JavaVersion/LaserBot_Galileo/src/laserbot_galileo/svgPlotter/Point.java
/*
 *  Copyright Nguyen Duong Kim Hao @ 2014
 */
package laserbot_galileo.svgPlotter;

/**
 *
 * @author KimHao
 */
public final class Point {

    public int x, y;

    public Point() {
    }

    public Point(int x, int y) {
        this.x = x;
        this.y = y;
    }

    public final int distance(Point p) {
        return (int) Math.sqrt((x - p.x) * (x - p.x) + (y - p.y) * (y - p.y));
    }

    @Override
    public final String toString() {
        return String.format("[%d, %d]", x, y);
    }

    public boolean equals(Point p) {
        return (x == p.x && y == p.y);
    }

    public final Point copy() {
        return new Point(x, y);
    }

}
